internal/state/rollback.go

```diff
@@ -65,10 +65,11 @@ func Rollback(bs BlockStore, ss Store, removeBlock bool) (int64, []byte, error)
 		return -1, nil, err
 	}
 
+	nextHeight := rollbackHeight + 1
```

Comment:
I know this is unrelated to this PR, but, why are we:

* loading the previous `LastValidatorSet` from `rollbackHeight`, and
* loading the previous `ConsensusParams`from `rollbackHeight + 1` (i.e., `invalidState.LastBlockHeight`)

@melekes what do you think? I can't find an explanation, and might be a similar bug lurking

Reply:
I'm working on #2283 to be able to test this (and repro the problem) using `e2e`. I still need to introduce a "rollback" perturbation to `e2e` functionality. Will get to when I'm back.

Reply:
@sergio-mena,  different definition by origin author,  you could check these code to compare  `LastValidatorSet` with `ConsensusParams`,
`LastValidatorSet`: https://github.com/cometbft/cometbft/blob/635d0b596c7300c98caa57e7aab26a7a4579ab97/internal/state/store.go#L229

`ConsensusParams`: https://github.com/cometbft/cometbft/blob/635d0b596c7300c98caa57e7aab26a7a4579ab97/internal/state/store.go#L233